src/main/java/org/candlepin/subscriptions/tally/SnapshotSummaryProducer.java

```diff
@@ -61,41 +67,59 @@ protected SnapshotSummaryProducer(
   public void produceTallySummaryMessages(Map<String, List<TallySnapshot>> newAndUpdatedSnapshots) {
     AtomicInteger totalTallies = new AtomicInteger();
     newAndUpdatedSnapshots.forEach(
-        (orgId, snapshots) ->
-            snapshots.stream()
-                .sorted(Comparator.comparing(TallySnapshot::getSnapshotDate))
-                .map(snapshot -> summaryMapper.mapSnapshots(orgId, List.of(snapshot)))
-                .forEach(
-                    summary -> {
-                      if (validateTallySummary(summary)) {
-                        kafkaRetryTemplate.execute(
-                            ctx ->
-                                tallySummaryKafkaTemplate.send(tallySummaryTopic, orgId, summary));
-                        totalTallies.getAndIncrement();
-                      }
-                    }));
+        (orgId, snapshots) -> {
+          // Filter snapshots, as we only deal with hourly, non Any fields
+          // and measurement types other than Total
+          // when we transmit the tally summary message to the BillableUsage component.
+          snapshots.stream()
+              .filter(this::filterByHourlyAndNotAnySnapshots)
+              .peek(this::removeTotalMeasurements)
```

Comment:
I never used the `peek` method like this. It might work, but as the API says: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/stream/Stream.html#peek(java.util.function.Consumer):
- `peek` is an intermediate operation (it will only be executed if are in the middle of other operations), so if in the future, we change the order of the operations, it might stop working. 
- "This method exists mainly to support debugging"

I think the appropiate operation is using the `map` to do this.

Reply:
yeah I had map there but, Intellij suggested to use peek instead. Let me change it to map again